pages/internals/[[...page]].jsx

```diff
@@ -18,25 +20,51 @@ export default function InternalsLayout(props) {
   )
 }
 
-export async function getStaticPaths() {
-  const paths = await generateStaticPaths({
-    navDataFile: NAV_DATA,
-    localContentDir: CONTENT_DIR,
-    product: PRODUCT,
-  })
-  return { paths, fallback: false }
-}
+const { getStaticPaths, getStaticProps } = getStaticGenerationFunctions(
+  process.env.IS_CONTENT_PREVIEW &&
+    process.env.PREVIEW_FROM_REPO === 'terraform'
+    ? {
+        strategy: 'fs',
+        basePath: BASE_ROUTE,
+        localContentDir: CONTENT_DIR,
+        navDataFile: NAV_DATA,
+        product: PRODUCT.slug,
+        githubFileUrl(filepath) {
+          // This path rewriting is meant for local preview from `terraform`.
+          filepath = filepath.replace('preview/', '')
+          return `https://github.com/hashicorp/${PRODUCT.slug}/blob/main/website/${filepath}`
+        },
+        remarkPlugins: (params) => [
+          () => {
+            const product = PRODUCT.slug
+            const version = process.env.CURRENT_GIT_BRANCH
+            return function transform(tree) {
+              visit(tree, 'image', (node) => {
+                const originalUrl = node.url
+                const asset = path.posix.join('website', originalUrl)
+                const url = new URL(
+                  'https://mktg-content-api.vercel.app/api/assets'
+                )
+                url.searchParams.append('asset', asset)
+                url.searchParams.append('version', version)
+                url.searchParams.append('product', product)
 
-export async function getStaticProps({ params }) {
-  const props = await generateStaticProps({
-    navDataFile: NAV_DATA,
-    localContentDir: CONTENT_DIR,
-    params,
-    product: PRODUCT,
-    githubFileUrl(path) {
-      const filepath = path.replace('content/', '')
-      return `https://github.com/hashicorp/${PRODUCT.slug}/blob/main/website/docs/${filepath}`
-    },
-  })
-  return { props }
-}
+                node.url = url.toString()
+                console.log(`Rewriting asset url for local preview:
+- Found: ${originalUrl}
+- Replaced with: ${node.url}
+
+If this is a net-new asset, you'll need to commit and push it to GitHub.`)
+              })
+            }
+          },
+        ],
+      }
```

Comment:
Maybe a reasonable time to abstract out this remark plugin into a shared module?

Reply:
Attempted to make a solution that worked for `cdktf`, `terraform`, as well as future users
- 7ce94a479

Reply:
renamed `assetPathBuilder` to `getAssetPathParts`